src/platforms/windows/daemon/windowssplittunnel.cpp

```diff
@@ -0,0 +1,535 @@
+    if(m_driver == INVALID_HANDLE_VALUE){
+         logger.log() << "Can't query State from non Opened Driver";
+        return STATE_UNKNOWN;
+    }
+    DWORD bytesReturned;
+    SIZE_T outBuffer;
+    bool ok = DeviceIoControl(m_driver, IOCTL_GET_STATE,
+        nullptr, 0, &outBuffer, sizeof(outBuffer), &bytesReturned,nullptr);
+    if(!ok){
+        WindowsCommons::windowsLog("getState response failure");
+        return STATE_UNKNOWN;
+    }
+    if(bytesReturned == 0){
+        WindowsCommons::windowsLog("getState response is empty");
+        return STATE_UNKNOWN;
+    }
+    return static_cast<DRIVER_STATE>(outBuffer);
+}
+
+
+std::vector<uint8_t> WindowsSplitTunnel::generateAppConfiguration(const QStringList& appPaths){
+    // Step 1: Calculate how much size the buffer will need
+    size_t cummulated_string_size =0;
+    for(auto const& path : appPaths){
+       cummulated_string_size += path.toStdWString().size()* sizeof (wchar_t);
+       logger.log() << path;
+    }
+    size_t bufferSize = sizeof(CONFIGURATION_HEADER)
+            + (sizeof(CONFIGURATION_ENTRY) * appPaths.size())
+            + cummulated_string_size;
+    std::vector<uint8_t> outBuffer(bufferSize);
+
+    auto header = (CONFIGURATION_HEADER*)&outBuffer[0];
+    auto entry = (CONFIGURATION_ENTRY*)(header + 1);
+
+    auto stringDest = &outBuffer[0] + sizeof(CONFIGURATION_HEADER)
+            + (sizeof(CONFIGURATION_ENTRY) * appPaths.size());
+
+    SIZE_T stringOffset = 0;
+
+    for (const auto &path : appPaths)
+    {
+        auto wstr= path.toStdWString();
+        auto cstr = wstr.c_str();
+        auto stringLength = wstr.size() * sizeof(wchar_t);
+
+        entry->ImageNameLength = (USHORT)stringLength;
+        entry->ImageNameOffset = stringOffset;
+
+        memcpy(stringDest, cstr , wstr.size());
+
+        ++entry;
+        stringDest += stringLength;
+        stringOffset += stringLength;
+    }
+
+    header->NumEntries = appPaths.length();
+    header->TotalLength = bufferSize;
+
+    return outBuffer;
+}
+
+
+std::vector<uint8_t> WindowsSplitTunnel::generateIPConfiguration(){
+    std::vector<uint8_t> out(sizeof(IP_ADDRESSES_CONFIG));
+
+    auto config = reinterpret_cast<IP_ADDRESSES_CONFIG*>(&out[0]);
+    /* How to Choose the Right Adapter:
+     * Windows will has their own thing Called metric to rate each connection
+     * and use the lowest-cost network adapter possible.
+     *
+     * So the VPN-Adapter starts with a metric of 0, making it the most optimal route
+     * making it always the default.
+     * So as the "outside" Connection we want the 2nd best metric.
+     *
+     * We can't get the exact number without using powershell but lucky us:
+     * From msdn (GetAdaptersAddresses):
+     * >Starting with Windows 10, the order in which adapters appear in
+     * >the list is determined by the IPv4 or IPv6 route metric.
+     *
+     * So  QNetworkInterface::allInterfaces() is perfectly sorted for us.
```

Comment:
is there a way to check this too programmatically?

Reply:
Yes there is... but it's wierd. 
Maybe as an alternative, we could offer a "select your split tunnel device" box?